src/allotropy/parsers/appbio_quantstudio_designandanalysis/appbio_quantstudio_designandanalysis_calculated_documents.py

```diff
@@ -0,0 +1,677 @@
+from collections.abc import Iterator
+from typing import Optional
+
+from allotropy.parsers.appbio_quantstudio.calculated_document import (
+    CalculatedDocument,
+    DataSource,
+)
+from allotropy.parsers.appbio_quantstudio.decorators import cache
+from allotropy.parsers.appbio_quantstudio.views import ViewData
+from allotropy.parsers.appbio_quantstudio_designandanalysis.appbio_quantstudio_designandanalysis_structure import (
+    WellItem,
+)
+from allotropy.parsers.utils.uuids import random_uuid_str
+
+
+@cache
+def build_quantity(well_item: WellItem) -> Optional[CalculatedDocument]:
+    if (quantity := well_item.result.quantity) is None:
+        return None
+
+    return CalculatedDocument(
+        uuid=random_uuid_str(),
+        name="quantity",
+        value=quantity,
+        data_sources=[
+            DataSource(feature="cycle threshold result", reference=well_item),
+            DataSource(feature="Y-intercept", reference=well_item),
+            DataSource(feature="Slope", reference=well_item),
+        ],
+    )
+
+
+def build_quantity_mean(
+    view_data: ViewData[WellItem], sample: str, target: str
+) -> Optional[CalculatedDocument]:
+    well_items = view_data.get_leaf_item(sample, target)
+    if (quantity_mean := well_items[0].result.quantity_mean) is None:
+        return None
+
+    data_sources = []
+    for well_item in well_items:
+        quantity_ref = build_quantity(well_item)
+        if quantity_ref is None:
+            return None
```

Comment:
Are we sure this is the correct behavior if even 1 well_item has null quantity? Should there be any more detailed logging if we get a document where `quantity_mean` above is not None, but a `quantity_ref`is?

Reply:
@nlangowitz thanks - we'll tackle this and the recommendation below in another PR.